### ISSUE TITLE
models/denset.py doesn't have any weight init!

### ISSUE THREAD

**Issue by skrish13** [pytorch/vision#232]
Please correct me if I'm wrong, but I don't seem to find any weight init in the densenet [code](https://github.com/pytorch/vision/blob/master/torchvision/models/densenet.py).

**Comment by soumith** [comment]
it uses the default pytorch weight initialization

**Comment by skrish13** [comment]
Ah right, sorry. Is it the same for BatchNorm and Linear ([bias](https://github.com/liuzhuang13/DenseNet/blob/master/models/densenet.lua#L161)?). Referring to official Torch code from [here](https://github.com/liuzhuang13/DenseNet/blob/master/models/densenet.lua#L153)

Generally used to seeing a init function, so was surprised :P

**Comment by soumith** [comment]
yes. looks like it uses the default initialization for everything.

**Comment by skrish13** [comment]
I guess I'm missing something.... Default init in pytorch for linear (both bias & weight) seems to be uniform of (-std,std) and for batchnorm(weight) it seems to be uniform. Isn't it different from their implementation? They make linear bias zero and batchnorm weight to be 1?

I've seen this seperately implemented in lot of places so assumed default is something different. ex: [models/resnet.py](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py)

**Comment by soumith** [comment]
@colesbury

**Comment by skrish13** [comment]
Any updates? @colesbury @soumith

**Comment by victorhcm** [comment]
@soumith if we port the weights from Caffe, do you think it could be included?

**Comment by skrish13** [comment]
@victorhcm what weights are you talking about?

@colesbury any updates?

**Comment by fmassa** [comment]
Yes, indeed, it looks like we are missing the weight initializations from the original implementation, which can be found [here](https://github.com/liuzhuang13/DenseNet/blob/master/models/densenet.lua#L138-L156).

I believe this needs to be addressed, thanks for raising the issue

**Comment by alykhantejani** [comment]
Fixed in https://github.com/pytorch/vision/pull/368
